src/OptionsToken.sol

```diff
@@ -3,41 +3,32 @@ pragma solidity ^0.8.13;
 
 import {Owned} from "solmate/auth/Owned.sol";
 import {ERC20} from "solmate/tokens/ERC20.sol";
-import {SafeTransferLib} from "solmate/utils/SafeTransferLib.sol";
-import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
 
+import {IOptionsToken} from "./interfaces/IOptionsToken.sol";
 import {IOracle} from "./interfaces/IOracle.sol";
 import {IERC20Mintable} from "./interfaces/IERC20Mintable.sol";
+import {IExercise} from "./interfaces/IExercise.sol";
 
 /// @title Options Token
 /// @author zefram.eth
-/// @notice Options token representing the right to purchase the underlying token
-/// at an oracle-specified rate. Similar to call options but with a variable strike
-/// price that's always at a certain discount to the market price.
-/// @dev Assumes the underlying token and the payment token both use 18 decimals.
-contract OptionsToken is ERC20, Owned, IERC20Mintable {
```

Comment:
consider making this contract upgradeable to reduce friction when dealing with the unexpected